src/shared/dissect-image.c

```diff
@@ -116,6 +116,121 @@ static bool device_is_block(sd_device *d) {
 
         return streq(ss, "block");
 }
+
+static int enumerator_for_parent(sd_device *d, sd_device_enumerator **ret) {
+        _cleanup_(sd_device_enumerator_unrefp) sd_device_enumerator *e = NULL;
+        int r;
+
+        r = sd_device_enumerator_new(&e);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_allow_uninitialized(e);
+        if (r < 0)
+                return r;
+
+        r = sd_device_enumerator_add_match_parent(e, d);
+        if (r < 0)
+                return r;
+
+        *ret = TAKE_PTR(e);
+        return 0;
+}
+
+/* how many times to wait for the device nodes to appear */
+#define N_DEVICE_NODE_LIST_ATTEMPTS 10
+
+static int wait_for_partitions_to_appear(
+                int fd,
+                sd_device *d,
+                int num_partitions,
```

Comment:
hmm, I think this type should be size_t or unsigned or so, not int here. I mean, in the interest of encapsulation I think this new function shouldn't have to care that this function is going to be called with num_partitions determined from blkid_partlist_numof_partitions() which happens to return an int. I mean, the number of partitions cannot be negative, and hence we shouldn't allow it here.